llm/rag/README.md

```diff
@@ -0,0 +1,83 @@
+# Retrieval Augmented Generation with DeepSeek R1
+
+<p align="center">
+<img src="https://i.imgur.com/fI6Mzpi.png" alt="RAG with DeepSeek R1" style="width: 70%;">
+</p>
+
+## Large-Scale Legal Document Search and Analysis
+As legal document collections grow, traditional keyword search becomes insufficient for finding relevant information. Retrieval-Augmented Generation (RAG) combines the power of vector search with large language models to enable semantic search and intelligent answers.
+
+In particular:
+
+* **Accuracy**: RAG systems ground their responses in source documents, reducing hallucination and improving answer reliability.
+* **Context-Awareness**: By retrieving relevant documents before generating answers, the system provides responses that consider specific legal contexts.
+* **Traceability**: All generated answers can be traced back to source documents, crucial for legal applications.
+
+SkyPilot streamlines the deployment of RAG systems in the cloud by managing infrastructure and enabling efficient, cost-effective compute resource usage. We use [Alibaba-NLP/gte-Qwen2-7B-instruct] for generating document embeddings and distilled Deepseek R1 ([deepseek-ai/DeepSeek-R1-Distill-Llama-8B]) for generating final anwsers. 
+
+In this example, we use legal documents by [pile of law](https://huggingface.co/datasets/pile-of-law/pile-of-law) as example data to demonstrate RAG capabilities. The system processes a collection of legal texts, including case law, statutes, and legal discussions, to enable semantic search and intelligent question answering. This approach can help legal professionals quickly find relevant precedents, analyze complex legal scenarios, and extract insights from large document collections. 
+
```

Comment:
Would be good to have an overview figure here for showing the three stages, and a short description for how RAG is built. : )

Reply:
Also, would be good to have a `Why use SkyPilot` section